src/components/ids-slider/ids-slider.js

```diff
@@ -0,0 +1,1052 @@
+
+      if (this.isRtl) { trans *= -1; }
+      const transString = this.vertical ? `translate(0, ${trans}px)` : `translate(${trans}px, 0)`;
+      thumbDraggable.style.transform = transString;
+    }
+  }
+
+  /**
+   * Calculate the true value based on the percent value
+   * @param {number} percent the percent value to convert to numerical value btw min and max
+   * @returns {number} the calculated value
+   */
+  #calcValueFromPercent(percent) {
+    return ((percent / 100) * (this.max - this.min)) + (this.min);
+  }
+
+  /**
+   * Calculate the pixels to translate thumb(s) or progress track based on percent value
+   * @param {number} nStart the starting x or y coordinate of the slider
+   * @param {number} nEnd the ending x or y coordinate of the slider
+   * @param {number} percent the percent/location of the thumb relative to the slider
+   * @param {boolean} notCentered whether or not the thumb(s) or progress track is centered or not
+   * if notCentered is true, it will translate from 0
+   * if notCentered is false, it will translate negatively and positively with 0 being the center
+   * @returns {number} coordinates or the amount of pixels to translate by
+   */
+  #calcTranslateFromPercent(nStart, nEnd, percent, notCentered) {
+    // minus thumb height bc it overshoots
+    const editedRange = Math.abs(nEnd - nStart) - this.thumbDraggable.clientWidth;
+    let coord = (Math.ceil(percent) / 100) * editedRange;
+    coord = !notCentered ? coord - (editedRange / 2) : coord;
+
+    return coord;
+  }
+
+  /**
+   * Calculate the percent value based on mouse click location
+   * @param {number} n the mouse x or y coordinate
+   * @param {number} nStart the starting x or y coordinate of the slider
+   * @param {number} nEnd the ending x or y coordinate of the slider
+   * @param {number} thumbWidth the width of the thumb
+   * @returns {number} the percent/location of the thumb relative to the slider
+   */
+  #calcPercentFromMousePos(n, nStart, nEnd, thumbWidth) {
+    let percent = 0;
+    // allow bigger hit areas for controlling thumb
+    /* istanbul ignore next */
+    const range = Math.abs(nStart - nEnd) - thumbWidth / 2;
+    const endDelta = Math.abs(n - nEnd);
+    const startDelta = Math.abs(n - nStart);
+    if (endDelta > range) {
+      percent = 0;
+    } else if (startDelta > range) {
+      percent = 100;
+    } else {
+      percent = ((startDelta - thumbWidth / 2) / (range - thumbWidth / 2)) * 100;
+    }
+    return percent;
+  }
+
+  /** Attach all the necessary event listeners */
+  #attachEventListeners() {
+    // CHECK IF RTL
+    this.#addRtlListener();
+
+    // INIT AFTER CSS LOADS
+    this.#postRenderInitialization();
+
+    // RESIZE OBSERVER for when window size changes
+    this.#addResizeObserver();
+
+    // DRAGGABLE EVENTS
+    this.#addDragEvents();
+    this.type === 'double' && this.#addDragEvents('secondary');
+
+    // KEYBOARD EVENTS
+    this.#addKeyboardEvents();
```

Comment:
```suggestion
    this.#addKeyboarListeners();
```

Did this on the other PR... for a name

Reply:
I'll commit it on my end since it's not spelled correctly here

Reply:
Should this be `attachKeyboardListeners()` instead of `add`?

Reply:
Yes my bad as per https://github.com/infor-design/enterprise-wc/pull/321#issue-723790999 

attachEventHandlers and attachKeyboardListeners